### PR TITLE
Sorted key path improvements

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [ ] New extensions are written in Swift 5.0.
- [ ] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [ ] I have added tests for new extensions, and they passed.
- [ ] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [ ] All extensions are declared as **public**.
- [ ] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/SwiftStdlib/MutableCollectionExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
+//  MutableCollectionExtensions.swift
+//  SwifterSwift
+//
+//  Created by Luciano Almeida on 21/09/19.
+//  Copyright © 2019 SwifterSwift
+//
+
+public extension MutableCollection where Self: RandomAccessCollection {
+  /// SwifterSwift: Sort the collection  based on a keypath and a compare function.
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/MutableCollectionExtensions.swift]: Only 2 spaces for tabs here, unlike the rest of the project

**Comment by LucianoPAlmeida** [comment]
> You've deprecated functions that allow you to just call `sort(by: someKeyPath)` but the replacements don't allow for the same - you _must_ supply a comparison function.
> ```

Maybe we can, but the problem in allowing call it like that, is since we only deprecate the old one calling like that `sort(by: someKeyPath)`, the compiler would have two overloads to choose from the old `sort(by:ascending)` and `sort(by:with:)` which can lead to ambiguous error or it chooses the deprecated one.

**Comment by guykogus** [comment]
> Maybe we can, but the problem in allowing call it like that, is since we only deprecate the old one calling like that `sort(by: someKeyPath)`, the compiler would have two overloads to choose from the old `sort(by:ascending)` and `sort(by:with:)` which can lead to ambiguous error or it chooses the deprecated one.

If there are overloads then delete the old ones that we don't need anymore, no?

**Comment by LucianoPAlmeida** [comment]
> If there are overloads then delete the old ones that we don't need anymore, no?
>

Maybe, since it would not be a breaking change :)

**Comment by LucianoPAlmeida** [comment]
> You've removed the ascending parameter, which means that you're introducing a breaking change.
>

I just removed the default value so that method can't be called like `sort(by: )` anymore, which is not a breaking change because those will be now calling the `sort(by:)` with the default comparator overload.

**Pull request merged**